src/inmanta/data/model.py

```diff
@@ -755,23 +767,44 @@ class DiscoveredResource(BaseModel):
         managed by the orchestrator e.g. "/api/v2/resource/<rid>". Or None if the resource is not managed.
     """
 
-    discovered_resource_id: ResourceIdStr
+    discovered_resource_id: ResourceId
     values: dict[str, object]
     managed_resource_uri: Optional[str] = None
 
-    @field_validator("discovered_resource_id")
-    @classmethod
-    def discovered_resource_id_is_resource_id(cls, v: str) -> str:
-        if resources.Id.is_resource_id(v):
-            return v
-        raise ValueError(f"id {v} is not of type ResourceIdStr")
+    _validate_discovered_rid = field_validator("discovered_resource_id")(is_resource_id)
```

Comment:
Aren't we now performing this validation twice. Once because of the type validation `ResourceId` and once here by setting the `_validate_discovered_rid` field.